lua/wire/zvm/zvm_core.lua

```diff
@@ -336,8 +336,45 @@ function ZVM:Dyn_EndBlock()
   return self.EmitBlock
 end
 
+--------------------------------------------------------------------------------
+-- Begins a block of code that will only run if we're in a quota supporting environment
+-- and about to exit due to quota.
+function ZVM:Dyn_BeginQuotaOnlyCode()
+  self:Dyn_Emit("if VM.QuotaSupported then")
+    self:Dyn_Emit("if VM.TMR > VM.Quota then")
+end
 
+--------------------------------------------------------------------------------
+-- Ends a block of "Quota only" code.
+function ZVM:Dyn_EndQuotaOnlyCode()
+    self:Dyn_Emit("end")
+  self:Dyn_Emit("end")
```

Comment:
Do we want to also include an
```
else
    error("Quota not supported")
end
```
?

Reply:
I think my intent was that inside of a context that doesn't (yet) support quotas, it'll behave like usual without exiting, but in a context that does, it can do the quota code and exit.